### ISSUE TITLE
PAF on Centos 7

### ISSUE THREAD

**Issue by alexseb76** [dalibo/PAF#85]
Hi all,
I have a big problem. My environment is based on 2 Centos 7 nodes  and postgresql 9.6. I have implemented the following procedure: https://dalibo.github.io/PAF/Quick_Start-CentOS-7.html.
After I created the resources, both server reboot and go in loop.
The error is: pgsqld_monitor_0 on pgsql1-cls 'invalid parameter' (2): call=18, status=complete, exitreason='none',
    last-rc-change='Tue Mar 14 11:32:51 2017', queued=0ms, exec=969ms

please, help me
Thanks a lot
Alessandro

**Comment by ioguix** [comment]
Hi,

You probably have a PostgreSQL setup error, either in `postgresql.conf` file or in the `recovery.conf.pcmk`.

* put your cluster in maintenance mode
* disable pacemaker on boot

check your logs using:

~~~
grep ^pgsqlms /var/log/cluster/corosync.log
~~~

Fix whatever errors you will find in your logs, do some cleanup/clear, exit from the maintenance mode.

~~~
pcs cleanup pgsql-ha
pcs clear pgsql-ha
pcs property set maintenance-mode=false
~~~

**Comment by alexseb76** [comment]
Hi,

thanks a lot for your support.

 

Will I set the cluster in maintenance mode on both nodes?

 

Regards

Alessandro

 

Da: Jehan-Guillaume (ioguix) de Rorthais [mailto:notifications@github.com] 
Inviato: martedì 14 marzo 2017 17:41
A: dalibo/PAF
Cc: Alessandro Sebastiano; Author
Oggetto: Re: [dalibo/PAF] PAF on Centos 7 (#85)

 

Hi,

You probably have a PostgreSQL setup error, either in postgresql.conf file or in the recovery.conf.pcmk.

*	put your cluster in maintenance mode
*	disable pacemaker on boot

check your logs using:

grep ^pgsqlms /var/log/cluster/corosync.log

Fix whatever errors you will find in your logs, do some cleanup/clear, exit from the maintenance mode.

pcs cleanup pgsql-ha
pcs clear pgsql-ha
pcs property set maintenance-mode=false

**Comment by ioguix** [comment]
> Will I set the cluster in maintenance mode on both nodes? 

I'm not sure to understand your question, but I'll try to answer. You do not need to execute the following command on all nodes, it is propagated everywhere automatically. You will see your cluster goes in maintenance mode in `crm_mon`.

~~~
pcs property set maintenance-mode=false
~~~

**Comment by alexseb76** [comment]
Hi,

I executed your instructions:

                set in maintenance mode

                re-create the postgresql resources

 

the errori I found:

pgsqlms(pgsqld)[24177]: 2017/03/17_14:47:19  CRIT: Recovery template file must contain in primary_conninfo parameter "application_name=pgsql1-cls"

 

I have on both nodes 2 network interfaces

Pgsql1   10.4.29.121

Pgsql-cls 192.168.1.121

 

Regards

Alessandro

 

Da: Jehan-Guillaume (ioguix) de Rorthais [mailto:notifications@github.com] 
Inviato: venerdì 17 marzo 2017 11:16
A: dalibo/PAF
Cc: Alessandro Sebastiano; Author
Oggetto: Re: [dalibo/PAF] PAF on Centos 7 (#85)

 

	Will I set the cluster in maintenance mode on both nodes?

I'm not sure to understand your question, but I'll try to answer. You do not need to execute the following command on all nodes, it is propagated everywhere automatically. You will see your cluster goes in maintenance mode in crm_mon.

pcs property set maintenance-mode=false

**Comment by ioguix** [comment]
Well, the error message is quite clear. Please, read the documentation and check everything twice.

http://dalibo.github.io/PAF/configuration.html

PS: could you clean the email history while answering the thread by email? This is quite hard to follow the discussion from the github interface. Thank you.

**Comment by alexseb76** [comment]
Hi,

now I changed the application name with pgsql1-cls and the node not reboot after the postgresql resources creation

 

Now, when I try to switch the resources on another node, the VIP not switch.

 

The warning in the log is 

WARNING: _check_locations: "pgsql2-cls" is not connected to the primary, set score to -1000

 

Please help me 

 

Thansk a lot

Alessandro

 

Da: Alessandro Sebastiano 
Inviato: venerdì 17 marzo 2017 14:53
A: 'dalibo/PAF'; dalibo/PAF
Cc: Author
Oggetto: R: [dalibo/PAF] PAF on Centos 7 (#85)

 

Hi,

I executed your instructions:

                set in maintenance mode

                re-create the postgresql resources

 

the errori I found:

pgsqlms(pgsqld)[24177]: 2017/03/17_14:47:19  CRIT: Recovery template file must contain in primary_conninfo parameter "application_name=pgsql1-cls"

 

I have on both nodes 2 network interfaces

Pgsql1   10.4.29.121

Pgsql-cls 192.168.1.121

 

Regards

Alessandro

 

Da: Jehan-Guillaume (ioguix) de Rorthais [mailto:notifications@github.com] 
Inviato: venerdì 17 marzo 2017 11:16
A: dalibo/PAF
Cc: Alessandro Sebastiano; Author
Oggetto: Re: [dalibo/PAF] PAF on Centos 7 (#85)

 

	Will I set the cluster in maintenance mode on both nodes?

I'm not sure to understand your question, but I'll try to answer. You do not need to execute the following command on all nodes, it is propagated everywhere automatically. You will see your cluster goes in maintenance mode in crm_mon.

pcs property set maintenance-mode=false

**Comment by alexseb76** [comment]
Hi All,
I have solved, with your support" the error message: pgsqlms(pgsqld)[24177]: 2017/03/17_14:47:19  CRIT: Recovery template file must contain in primary_conninfo parameter "application_name=pgsql1-cls"

I set in the recovery file the name in the host file of the heartbeat.

Now if I power-off one node the Virtual IP no switch on another node.

Could you help me?
Thanks a lot
Alessandro

**Comment by blogh** [comment]
Hi @alexseb76,

Could you show us your pcs config  please ?

Benoit.

**Comment by alexseb76** [comment]
Benoit,
how can I export the configuration?
Thanks a lot

**Comment by alexseb76** [comment]
Hi,
I have killed the corosync but the VIP no switch
 pgsql-master-ip        (ocf::heartbeat:IPaddr2):       Stopped

**Comment by alexseb76** [comment]
this is the result of pcs status
Cluster name: pgcluster
Stack: corosync
Current DC: pgsql2-cls (version 1.1.15-11.el7_3.4-e174ec8) - partition with quorum
Last updated: Mon Mar 20 14:29:01 2017          Last change: Fri Mar 17 16:01:12 2017 by root via crm_resource on pgsql1-cls

2 nodes and 5 resources configured

Online: [ pgsql1-cls pgsql2-cls ]

Full list of resources:

 pgsql-master-ip        (ocf::heartbeat:IPaddr2):       Started pgsql1-cls
 fence_pgsql1   (stonith:fence_vmware_soap):    Started pgsql1-cls
 fence_pgsql2   (stonith:fence_vmware_soap):    Started pgsql2-cls
 Master/Slave Set: pgsql-ha [pgsqld]
     Masters: [ pgsql1-cls ]
     Stopped: [ pgsql2-cls ]

Failed Actions:
* pgsqld_monitor_0 on pgsql2-cls 'not installed' (5): call=18, status=Not installed, exitreason='none',
    last-rc-change='Fri Mar 17 15:59:51 2017', queued=0ms, exec=0ms
* pgsqld_monitor_0 on pgsql1-cls 'master (failed)' (9): call=18, status=complete, exitreason='none',
    last-rc-change='Mon Mar 20 14:23:30 2017', queued=0ms, exec=997ms


I think there is a problem with stopped second node
Master/Slave Set: pgsql-ha [pgsqld]
     Masters: [ pgsql1-cls ]
     Stopped: [ pgsql2-cls ]

Why?

**Comment by ioguix** [comment]
> Why ?

Se in your log files...You probably have a line that explains why the monitor action raised a 'not installed' error. As I already wrote:

~~~
grep ^pgsqlms /var/log/cluster/corosync.log
~~~

**Comment by alexseb76** [comment]
I have this warning
pgsqlms(pgsqld)[16369]: 2017/03/20_14:56:31  WARNING: _check_locations: "pgsql2-cls" is not connected to the primary, set score to -1000

**Comment by ioguix** [comment]
This is not related to the "not configured" error you see on pgsql2-cls. Please, look at the log files ont on pgsql2-cls...

**Comment by alexseb76** [comment]
I killed on node 1  but on pgsql2-cls the result of grep ^pgsqlms /var/log/cluster/corosync.log
is empty.

**Comment by ioguix** [comment]
Look, based on what you gave earlier, you had the following error __on pgsql2-cls__:

~~~
pgsqld_monitor_0 on pgsql2-cls 'not installed' (5): call=18, status=Not installed, exitreason='none',
last-rc-change='Fri Mar 17 15:59:51 2017', queued=0ms, exec=0ms
~~~

The "not installed" is raised by PAF itself the __Fri Mar 17 15:59:51__ and it usually shout in the log files what was the error. So you should have an error logged somewhere in a file of pgsql2-cls. Make sure your log files hasn't been rotated, look for the files logging data for the 17th.

**Comment by alexseb76** [comment]
One of the problem is that the paf resource agent was not installed on node 2.
Now the pcs status is:
2 nodes and 5 resources configured

Online: [ pgsql1-cls pgsql2-cls ]

Full list of resources:

 pgsql-master-ip        (ocf::heartbeat:IPaddr2):       Started pgsql2-cls
 fence_pgsql1   (stonith:fence_vmware_soap):    Started pgsql1-cls
 fence_pgsql2   (stonith:fence_vmware_soap):    Started pgsql2-cls
 Master/Slave Set: pgsql-ha [pgsqld]
     Masters: [ pgsql2-cls ]
     Slaves: [ pgsql1-cls ]

Daemon Status:
  corosync: active/enabled
  pacemaker: active/enabled
  pcsd: active/enabled

I killed corosync process on the primary node slave PGSQL2-CLS but the IP no switch on node 1.

The log on the PGQL1-CLS  (SLAVE) is
pgsqlms(pgsqld)[8794]:  2017/03/20_16:56:26  INFO: pgsql_notify: promoting instance on node "pgsql2-cls"
pgsqlms(pgsqld)[8794]:  2017/03/20_16:56:26  INFO: pgsql_notify: current node LSN: 0/3000000
pgsqlms(pgsqld)[9154]:  2017/03/20_16:58:47  INFO: pgsql_notify: promoting instance on node "pgsql2-cls"
pgsqlms(pgsqld)[9154]:  2017/03/20_16:58:47  WARNING: pgsql_notify: This is a master recovery!

The log on the PGSQL2-CLS (PRIMARY) is
pgsqlms(pgsqld)[2539]:  2017/03/20_16:56:28  INFO: pgsql_start: instance "pgsqld" started
pgsqlms(pgsqld)[2586]:  2017/03/20_16:56:28  INFO: pgsql_notify: promoting instance on node "pgsql2-cls"
pgsqlms(pgsqld)[2586]:  2017/03/20_16:56:28  INFO: pgsql_notify: current node LSN: 0/3000000
pgsqlms(pgsqld)[2611]:  2017/03/20_16:56:29  INFO: pgsql_promote: promote complete
pgsqlms(pgsqld)[2660]:  2017/03/20_16:56:30  WARNING: _check_locations: No secondary connected
pgsqlms(pgsqld)[2660]:  2017/03/20_16:56:30  WARNING: _check_locations: "pgsql1-cls" is not connected to the primary, set score to -1000
pgsqlms(pgsqld)[2918]:  2017/03/20_16:56:45  WARNING: _check_locations: No secondary connected
pgsqlms(pgsqld)[2918]:  2017/03/20_16:56:45  WARNING: _check_locations: "pgsql1-cls" is not connected to the primary, set score to -1000
pgsqlms(pgsqld)[2545]:  2017/03/20_16:58:33  ERROR: _confirm_stopped: instance "pgsqld" controldata indicates a running primary instance, the instance has probably crashed
pgsqlms(pgsqld)[2585]:  2017/03/20_16:58:42  ERROR: _confirm_stopped: instance "pgsqld" controldata indicates a running primary instance, the instance has probably crashed
pgsqlms(pgsqld)[2585]:  2017/03/20_16:58:42  INFO: pgsql_notify: trying to start failing master "pgsqld"...
pgsqlms(pgsqld)[2585]:  2017/03/20_16:58:43  INFO: pgsql_notify: state is "in production" after recovery attempt
pgsqlms(pgsqld)[2620]:  2017/03/20_16:58:45  INFO: pgsql_start: instance "pgsqld" started
pgsqlms(pgsqld)[2620]:  2017/03/20_16:58:45  INFO: pgsql_demote: "pgsqld" started as a secondary
pgsqlms(pgsqld)[2691]:  2017/03/20_16:58:47  INFO: pgsql_stop: instance "pgsqld" stopped
pgsqlms(pgsqld)[2710]:  2017/03/20_16:58:48  INFO: pgsql_start: instance "pgsqld" started
pgsqlms(pgsqld)[2756]:  2017/03/20_16:58:48  INFO: pgsql_notify: promoting instance on node "pgsql2-cls"
pgsqlms(pgsqld)[2756]:  2017/03/20_16:58:48  WARNING: pgsql_notify: This is a master recovery!
pgsqlms(pgsqld)[2769]:  2017/03/20_16:58:48  INFO: pgsql_promote: recovering old master, no election needed
pgsqlms(pgsqld)[2769]:  2017/03/20_16:58:53  INFO: pgsql_promote: promote complete
pgsqlms(pgsqld)[2831]:  2017/03/20_16:58:54  WARNING: _check_locations: No secondary connected
pgsqlms(pgsqld)[2831]:  2017/03/20_16:58:54  WARNING: _check_locations: "pgsql1-cls" is not connected to the primary, set score to -1000

**Comment by alexseb76** [comment]
Hi all,
who can help me? I can send us log file, configuration.
I need to install on the customer side. Now the situation is that the VIP not switch on the second node.

many thanks
Alessandro

**Comment by alexseb76** [comment]
Just to understand where is the problem...when I set the last constraints I receive this message:

`pcs constraint order promote pgsql-ha then start pgsql-master-ip symmetrical=false`
promote pgsql-ha then start pgsql-master-ip (kind:Mandatory) (non-symmetrical) (id:order-pgsql-ha-pgsql-master-ip-mandatory)

`pcs constraint order demote pgsql-ha then stop pgsql-master-ip symmetrical=false`
demote pgsql-ha then stop pgsql-master-ip (kind:Mandatory) (non-symmetrical) (id:order-pgsql-ha-pgsql-master-ip-mandatory-1)

Is it a problem with syntax?
Regards
Alessandro

**Comment by ioguix** [comment]
According to you log files given [in this message](https://github.com/dalibo/PAF/issues/85#issuecomment-287806135), you had a failover to PGSQL2-CLS as the PostgreSQL master is on this node.

However, I don't know what is happening on your node, but according to the same log file, this master is seen as crashed 2 minutes later and recovered.

I see no errors in your location constraints.

**Comment by alexseb76** [comment]
Now the scenario is that wher I power-off the master PGSQL2-CLS the VIP switch on PGSQL1-CLS node, but the old master now reboot Always with this error message on pcs status

Failed Actions:
* pgsqld_monitor_0 on pgsql2-cls 'invalid parameter' (2): call=18, status=complete, exitreason='none',
    last-rc-change='Wed Mar 22 14:45:00 2017', queued=0ms, exec=839ms

Failed Actions:
* pgsqld_stop_0 on pgsql2-cls 'invalid parameter' (2): call=22, status=complete, exitreason='none',
    last-rc-change='Wed Mar 22 14:45:09 2017', queued=0ms, exec=85ms

**Comment by ioguix** [comment]
As usual...look at your logs...

Note that Pacemaker/Corosync shouldn't start on boot, as stated in the documentation.

**Comment by alexseb76** [comment]
So, Do I have start the pacemaker/corosync on boot?

**Comment by ioguix** [comment]
I just don't understand your question, sorry.

**Comment by alexseb76** [comment]
Please, I want  to understand the reasons of my issues. I followed all steps described in the procedure 
https://dalibo.github.io/PAF/Quick_Start-CentOS-7.html.

I want to solve this error message 

Failed Actions:
pgsqld_monitor_0 on pgsql2-cls 'invalid parameter' (2): call=18, status=complete, exitreason='none',
last-rc-change='Wed Mar 22 14:45:00 2017', queued=0ms, exec=839ms
Failed Actions:
pgsqld_stop_0 on pgsql2-cls 'invalid parameter' (2): call=22, status=complete, exitreason='none',
last-rc-change='Wed Mar 22 14:45:09 2017', queued=0ms, exec=85ms

Have I make additional steps? 
Many thanks

**Comment by alexseb76** [comment]
News!! 
I found an invalid parameter in the postgres resource. 
Now when I restart the node, the service switch normally on the second node. But, when I poweroff from hypervisor one node, this reboot in loop and I don't know hao can restore this node.

the error message is
Failed Actions:
* pgsqld_monitor_0 on pgsql1-cls 'master (failed)' (9): call=18, status=complete, exitreason='none',

Please, help me!!!

**Comment by ioguix** [comment]
If it reboots in loop, that means Pacemaker is starting on boot.

As I already wrote: disable Pacemaker and Corosync on boot.

**Comment by alexseb76** [comment]
ok I disabled the services on boot on both nodes. The VIP switchs normally on second node after pskill -9 corosync on PGSQL1. 
Now the status is that the first node is offline

Online: [ pgsql2-cls ]
OFFLINE: [ pgsql1-cls ]

Full list of resources:

 pgsql-master-ip        (ocf::heartbeat:IPaddr2):       Started pgsql2-cls
 fence_pgsql1   (stonith:fence_vmware_soap):    Started pgsql2-cls
 fence_pgsql2   (stonith:fence_vmware_soap):    Started pgsql2-cls
 Master/Slave Set: pgsql-ha [pgsqld]
     Masters: [ pgsql2-cls ]
     Stopped: [ pgsql1-cls ]


How can make online the first node after failover simulation?
Thanks a lot

**Comment by ioguix** [comment]
> How can make online the first node after failover simulation?

Reboot it. Fix everything on it and start Pacemaker by hands on it.

Please, read the doc, the Pacemaker one and the PAF one. And pay attention to your logs. 

I repeated here many times "read the doc" and "read your logs", not for fun, but because most of your questions were already covered there.

Moreover, I already wrote yesterday: "Note that Pacemaker/Corosync shouldn't start on boot, as stated in the documentation". In regard with our discussion today, it seems you just did it today and your last question was exactly because of this...

It's hard to help you. Try to make it easier for people trying to help you to answer to your questions.

Thank you.

**Comment by alexseb76** [comment]
Last question.
I have to install 2 virtual machine on Microsoft hyper-v 2012 R2. Can I create stonith fencing logging hyper-v?

Thanks a lot

**Comment by ioguix** [comment]
Hi,

I don't know, I never worked with such virtual environment. Maybe you should ask on the clusterlabs user mailing list: http://lists.clusterlabs.org/mailman/listinfo/users
